### DISCUSSION TITLE
Support UNIX domain sockets for VNC backend(?)

### DISCUSSION THREAD

**Discussion by matt335672** [neutrinolabs/xrdp#2234] [General]
This discussion is prompted by a [comment](https://github.com/neutrinolabs/xrdp/pull/2207#issuecomment-1102636166) from @Nexarian.

It seems that the TigerVNC X server can support connections over UNIX Domain Sockets as well as the standard RFB port (5900+$DISPLAY). See [this discussion on stackoverflow](https://stackoverflow.com/questions/11314866/vncserver-supports-unix-domain-sockets).

Is this something we would want to actively support? My own feeling is probably not, as it's TigerVNC-specific and relatively new, but I might be missing something useful here.

**Comment by kbucheli** [comment]
Would this make following scenario possible, where on desktop session startup a password-less VNC server would be autostarted, but only accessible over a socket in a place where only the user has access? And xrdp/sesman would do PAM authentication of the user to ensure that only the specific user access to that socket?

possible xrdp.ini configuration:

```
[x0vncserver]
name=Local Desktop Session
lib=libvnc.@lib_extension@
socket=/run/user/$UID/vnc.socket
pamusername=ask
pampassword=ask
```

**Reply by matt335672** [reply]
Yes, I think it would - thanks.

Another use-case is in #2518 where DES is not available to provide authentication for VNC.
